contracts/UFragmentsPolicy.sol

```diff
@@ -340,7 +371,28 @@ contract UFragmentsPolicy is Ownable {
             rebaseFunctionUpperPercentage,
             rebaseFunctionGrowth
         );
-        return uFrags.totalSupply().toInt256Safe().mul(rebasePercentage).div(ONE);
+
+        int256 currentSupply = uFrags.totalSupply().toInt256Safe(); // (or) supplyHistory[epoch]
+        int256 newSupply = ONE.add(rebasePercentage).mul(currentSupply).div(ONE);
+
+        // When supply is decreasing:
+        // We limit the supply delta, based on recent supply history.
+        if (rebasePercentage < 0) {
+            int256 maxSupplyInHistory = currentSupply;
+            for (uint8 i = 1; i < epochLookback && epoch > i; i++) {
+                int256 epochSupply = supplyHistory[epoch - i].toInt256Safe();
+                if (epochSupply > maxSupplyInHistory) {
+                    maxSupplyInHistory = epochSupply;
+                }
+            }
+            int256 allowedSupplyMinimum = maxSupplyInHistory
+                .mul(ONE.sub(tolerableDeclinePercentage))
+                .div(ONE);
+            newSupply = (newSupply > allowedSupplyMinimum) ? newSupply : allowedSupplyMinimum;
+            require(newSupply <= currentSupply);
```

Comment:
This could cause the rebase operation to fail, which we've recently started to avoid.
Should we make this recover into a no-op supply change?

Reply:
We can get rid of this require. If the implementation is correct, `newSupply <= currentSupply` should always be true. Given that supply is reducing, the limiting logic should never increase the supply.